src/pybind/mgr/cephadm/services/monitoring.py

```diff
@@ -194,15 +194,20 @@ class PrometheusService(CephadmService):
     TYPE = 'prometheus'
     DEFAULT_SERVICE_PORT = 9095
 
+    def config(self, spec: ServiceSpec, daemon_id: str) -> None:
+        mgr_map = self.mgr.get('mgr_map')
+        if 'prometheus' not in mgr_map.get('services', {}):
+            self.mgr.check_mon_command({
+                'prefix': 'mgr module enable',
+                'module': 'prometheus'
+            })
```

Comment:
this might be too late, if the next call to generate_config does not contain the running mgr/prometheus. We might end up deploying the prometheus daemons twice. 

any ideas? I don't like `time.sleep(10)` as this is going to slow down everything.

Reply:
I hope I understood it correctly, so that my suggestion is actually useful. Would it really slow down anything if we just waited for the service to come up that was just activated? I mean if it's active, there's probably not much to wait for and if it isn't, there's reason to wait and even try again, isn't it?

Reply:
"try again" in the sense of checking for a result again, not necessarily to repeat the redeploy.